MFRC522.cpp

```diff
@@ -820,6 +820,62 @@ MFRC522::StatusCode MFRC522::PICC_HaltA() {
 	return result;
 } // End PICC_HaltA()
 
+/**
+ * Transmits a Request command for Answer To Select (ATS).
+ *
+ * @return STATUS_OK on success, STATUS_??? otherwise.
+ */
+MFRC522::StatusCode MFRC522::PICC_RATS(byte *bufferATS,	///< The buffer to store the ATS (Answer to select) in
```

Comment:
I think there are no changes apart from the name. In the DESFire library I choosed the long name, maybe more descriptive, and in the first versions I was using the abbreviature. 

Really the RATS is a bit more complex... In that method I'm using:

`bufferATS[1] = 0x50; // FSD=128, CID=0`

The byte sent represents the FSD and the CID. This is described here:
http://www.gorferay.com/data-transmission-protocol-iso-iec-14-433-4/

In the reply you would get additional information for the PPS command, which would allow to change the speed of the reader according to the parameters exchanged. I didn't want to go into tweaking the method too much as it may get a little confusing and not so sure if itt can bee nicelly representated (I'll look into it). Also, in my library I'm constantlly using CID 0 for the blocks, that is because CID = 0 has been hardcoded in the RATS method.

Maybe I could take a look into both those methods to get them to work with more flexibility... I was so focused on DESFire code I was happy with it doing both step to take me to the application layer DESFire. Recently I bumped into a wall :) I was testing all of this with a public transport card of my region and everything was working ok until the point I've hit encryption. I don't know the companies keys so I'm now waiting for some blank DESfire tags to arrive. I was also looking a little into 125 kHz cards... but it could be nice to find a way to finish off both of this methods completelly :) I'll look into them once more :)